src/pages/ReportDetailsPage.tsx

```diff
@@ -87,9 +87,12 @@ function ReportDetailsPage({policies, report, route, reportMetadata}: ReportDeta
     const backTo = route.params.backTo;
 
     // The app would crash due to subscribing to the entire report collection if parentReportID is an empty string. So we should have a fallback ID here.
-    const [parentReport] = useOnyx(`${ONYXKEYS.COLLECTION.REPORT}${report.parentReportID}`);
-    const [reportNameValuePairs] = useOnyx(`${ONYXKEYS.COLLECTION.REPORT_NAME_VALUE_PAIRS}${report?.reportID}`);
-    const [parentReportNameValuePairs] = useOnyx(`${ONYXKEYS.COLLECTION.REPORT_NAME_VALUE_PAIRS}${report?.parentReportID}`);
+    // eslint-disable-next-line rulesdir/no-default-id-values
+    const [parentReport] = useOnyx(`${ONYXKEYS.COLLECTION.REPORT}${report.parentReportID ?? ''}`);
+    // eslint-disable-next-line rulesdir/no-default-id-values
+    const [reportNameValuePairs] = useOnyx(`${ONYXKEYS.COLLECTION.REPORT_NAME_VALUE_PAIRS}${report?.reportID ?? ''}`);
```

Comment:
Should we use `CONST.DEFAULT_NUMBER_ID` as the defaults here? Or is `''` better?

Reply:
using `CONST.DEFAULT_NUMBER_ID` also breaks the app because key is dynamically changed from `report_` to `report_0`. I think using `''` is the only safe way here

Reply:
Interesting! I am surprised b/c we use `CONST.DEFAULT_NUMBER_ID` so often, even for the report ID like here: 

https://github.com/Expensify/App/blob/d641a12bd547739363a47d4ffededd5702b520aa/src/pages/home/report/ReportActionsView.tsx#L95

I am very ok with keeping this `''` for now, but I think we'll eventually want a better solution OR we should make eslint accept default to an empty string, what do you think? :D

Reply:
I think ideally in the future we could align the shape of optimistic report with the one that comes from `OpenReport` endpoint - we optimistically create it with `parentReportID` as empty string and then we receive `null` in a response. This might require a change on the backend

I'm tagging @VickyStash to answer allowing empty string as default question, as she has much more context of this rule than me 😅

Reply:
Innnteresting! Ok so how about for now, since your solution should work, let's get it merged & CP'd so we can unblock deploy, then we can figure out "the best solution" as a follow-up?

Reply:
It looks like somewhere in the app the `parentReportID` is optimistically set to empty string `''`.

So this way the hook 

```
const [parentReport] = useOnyx(`${ONYXKEYS.COLLECTION.REPORT}${report.parentReportID}`);
```
tries to get the data using firstly

`report_` key - the whole collection key
`report_undefined` key - specific report key

Since one `useOnyx` hook is used to get both collection and specific item it gives error.
For the same reason the nullish coalescing rule was disabled [before](https://github.com/Expensify/App/pull/54652/files#r1899490222).

Ideally, we should find out why `parentReportID` is turned into a string and prevent it. But we discussed it with @TMisiukiewicz and it looks like it can have side effects at this stage.
For now, we can also check specifically if `parentReportID` is not an empty string or [turn it back to as it was before](https://github.com/Expensify/App/pull/54652/files#r1899490222) (@shubham1206agra mentioned below).